docs/core/project-sdk/msbuild-props.md

```diff
@@ -842,6 +845,27 @@ The `DisableRuntimeMarshalling` property enables you to specify that you would l
 </PropertyGroup>
 ```
 
+### BuildWithNetFrameworkHostedCompiler
+
+When using .NET Framework MSBuild, `BuildWithNetFrameworkHostedCompiler=true` ensures that
+a C#/VB compiler corresponding to the current SDK version is used
+instead of the default version that ships with MSBuild.
+When this property is set to `true`, the .NET Framework version of the compiler is used, unlike `RoslynUseSdkCompiler`.
+In some cases, this behavior happens automatically when it is detected that MSBuild and SDK versions are different,
+and then you can set `BuildWithNetFrameworkHostedCompiler=false` to opt out of the behavior.
+
+### RoslynUseSdkCompiler
+
+When using .NET Framework MSBuild, `RoslynUseSdkCompiler=true` ensures that
+a C#/VB compiler corresponding to the current SDK version is used
+instead of the default version that ships with MSBuild.
+When this property is set to `true`, the .NET Core version of the compiler is used, unlike `BuildWithNetFrameworkHostedCompiler`.
+In most cases, `RoslynUseSdkCompiler=true` is the default setting.
+
+### RoslynUseMSBuildCompiler
+
+`RoslynUseMSBuildCompiler=true` can be used to opt out of an implicit `RoslynUseSdkCompiler=true`.
```

Comment:
Interesting. Why is this property necessary? Isn't setting `<RoslynUseSdkCompiler>false</RoslynUseSdkCompiler>` sufficient?

Reply:
Yes, it should be sufficient. RoslynUseMSBuildCompiler was documented by @jaredpar in https://github.com/dotnet/sdk/blob/main/documentation/general/decouple-vs-and-net-sdk.md that's why I included it in the implementation and here, but I agree it feels like we might not need it.